internal/metrics/metrics.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type MetricsHandler interface {
+	http.Handler
+}
+
+// PrometheusHandler implements the MetricsHandler interface.
+type PrometheusHandler struct {
+	registry *prometheus.Registry
+}
+
+// Returns a new PrometheusHandler.
+func NewPrometheusHandler() MetricsHandler {
+	registry := prometheus.NewRegistry()
+	registry.MustRegister(collectors.NewGoCollector())
+	return &PrometheusHandler{
+		registry: registry,
+	}
+}
+
+// ServeHTTP implements the http.Handler interface, allowing the PrometheusHandler to handle HTTP requests.
+func (p *PrometheusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	promhttp.HandlerFor(p.registry, promhttp.HandlerOpts{}).ServeHTTP(w, r)
+}
```

Comment:
Do we need all of this? With my recommendation above, we could have something like

```
func NewMetricsHandler() http.Handler {
	reg := prometheus.NewRegistry()
	reg.MustRegister(collectors.NewGoCollector())
	prometheusHandler := promhttp.HandlerFor(reg, promhttp.HandlerOpts{})
	return prometheusHandler
}
```

Reply:
What you are proposing would definitely be sufficient right now, however the aim was to have a way of implementing metrics handlers other than the one for Prometheus.
What do you think?

Reply:
Done